dm3-names/test/testDm3NameRegistrar.ts

```diff
@@ -0,0 +1,135 @@
+    });
+    it('can use reverse record to retrive name of address', async () => {
+      await target.register('alice');
+      const reverseRecord = `${signer.address
+        .slice(2)
+        .toLowerCase()}.addr.reverse`;
+      const reverseNode = ethers.namehash(reverseRecord);
+
+      const name = await target.name(reverseNode);
+      expect(name).to.equal('alice');
+    });
+    it('registering a new name overrides the old name', async () => {
+      await target.register('alice');
+      const reverseRecord = `${signer.address
+        .slice(2)
+        .toLowerCase()}.addr.reverse`;
+
+      let owner = await target.owner(ethers.namehash('alice.op.dm3.eth'));
+      let name = await target.reverse(ethers.namehash(reverseRecord));
+
+      expect(owner).to.equal(signer.address);
+      expect(name).to.equal('alice');
+
+      await target.register('bob');
+
+      owner = await target.owner(ethers.namehash('bob.op.dm3.eth'));
+      name = await target.reverse(ethers.namehash(reverseRecord));
+
+      const oldOwner = await target.owner(ethers.namehash('alice.op.dm3.eth'));
+
+      expect(owner).to.equal(signer.address);
+      expect(name).to.equal('bob');
+
+      expect(oldOwner).to.equal(ethers.ZeroAddress);
+    });
+    it('passing an empty name deletes an existing record', async () => {
+      await target.register('alice');
+      const reverseRecord = `${signer.address
+        .slice(2)
+        .toLowerCase()}.addr.reverse`;
+
+      let owner = await target.owner(ethers.namehash('alice.op.dm3.eth'));
+      let name = await target.reverse(ethers.namehash(reverseRecord));
+
+      expect(owner).to.equal(signer.address);
+      expect(name).to.equal('alice');
+
+      await target.register(ethers.toUtf8Bytes(''));
+
+      owner = await target.owner(ethers.namehash('alice.op.dm3.eth'));
+      name = await target.reverse(ethers.namehash(reverseRecord));
+
+      expect(owner).to.equal(ethers.ZeroAddress);
+      expect(name).to.equal('');
+    });
+  });
+
+  describe('setText', () => {
+    it('can set text record if name has been registered before', async () => {
+      await target.register('alice');
+      await target.setText(ethers.namehash('alice.op.dm3.eth'), 'key', 'value');
+      const value = await target.text(
+        ethers.namehash('alice.op.dm3.eth'),
+        'key'
+      );
+      expect(value).to.equal('value');
+    });
+
+    it('reverts if name has not been registered', async () => {
+      await expect(
+        target.setText(ethers.namehash('alice.op.dm3.eth'), 'key', 'value')
+      ).to.be.revertedWith('Name not registered');
+    });
+    it('reverts if msg.sender is not owner of name', async () => {
+      await target.register('alice');
+      const other = (await ethers.getSigners())[1];
+      await expect(
+        target
+          .connect(other)
+          .setText(ethers.namehash('alice.op.dm3.eth'), 'key', 'value')
+      ).to.be.revertedWith('Only owner');
+    });
```

Comment:
add test to make sure that I can not register "alex" after you already registered it.

Reply:
Todo add missing test